src/writeData/subscriptions/utils/form.ts

```diff
@@ -99,8 +99,8 @@ export const sanitizeForm = (form: Subscription): Subscription => {
       if (newVal) {
         f.path = newVal
       }
-      if (f.type === 'number') {
-        f.type = 'double'
+      if (f.type === 'integer') {
+        f.type = 'int'
```

Comment:
totally a future PR but I'm thinking we could refactor all of these `if` statements a bit. We could create an object that has the display value and the avro value as properties and have the user select one of those objects in the dropdown. Then instead of all these if statements we could just have the onChange event of the dropdown set `type` to the avro value or something. But anyways, a future PR :)

Reply:
100%